pkg/commands/start_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package commands
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	tcpPort = "5353"
+	udpPort = "5353"
+)
+
+func TestStart(t *testing.T) {
+	cmd := run(t, "127.0.0.1", tcpPort, udpPort, "host.rd.test=111.111.111.111,host2.rd.test=222.222.222.222")
+	defer cmd.Process.Kill()
+
+	t.Logf("Checking for TCP port is running on %v", tcpPort)
+	tcpListener, err := net.Listen("tcp", fmt.Sprintf(":%s", tcpPort))
+	if tcpListener != nil {
+		defer tcpListener.Close()
+	}
+	require.Errorf(t, err, "host-resolver is not listening on TCP port %s", tcpPort)
+
+	t.Logf("Checking for UDP port is running on %v", udpPort)
+	udpListener, err := net.Listen("udp", fmt.Sprintf(":%s", udpPort))
+	if udpListener != nil {
+		defer udpListener.Close()
+	}
+	require.Errorf(t, err, "host-resolver is not listening on UDP port %s", udpPort)
+
+	output := netstat(t)
+
+	exist := strings.Contains(string(output), fmt.Sprintf("%v/host-resolver", cmd.Process.Pid))
+	require.Exactly(t, exist, true, "Expected the same Pid")
+}
+
+func TestQueryTCP(t *testing.T) {
+	t.Logf("Checking for TCP port on %s", tcpPort)
+	cmd := run(t, "127.0.0.1", tcpPort, udpPort, "host.rd.test=111.111.111.111,host2.rd.test=222.222.222.222")
+	defer cmd.Process.Kill()
+
+	addrs, err := dnsLookup(t, tcpPort, "tcp", "host.rd.test")
+	require.NoError(t, err, "Lookup IP failed")
+	require.Exactly(t, len(addrs), 1, "Expect only one address")
+	require.Exactly(t, addrs[0].String(), "111.111.111.111")
+}
+
+func TestQueryUDP(t *testing.T) {
+	t.Logf("Checking for UDP port on %s", udpPort)
+	cmd := run(t, "127.0.0.1", tcpPort, udpPort, "host.rd.test=111.111.111.111,host2.rd.test=222.222.222.222")
+	defer cmd.Process.Kill()
+
+	addrs, err := dnsLookup(t, udpPort, "udp", "host2.rd.test")
+	require.NoError(t, err, "Lookup IP failed")
+	require.Exactly(t, len(addrs), 1, "Expect only one address")
+	require.Exactly(t, addrs[0].String(), "222.222.222.222")
+}
+
+func run(t *testing.T, ip, tcpPort, udpPort, hosts string) *exec.Cmd {
+	cmd := exec.Command("/host-resolver", "run", "-a", ip, "-t", tcpPort, "-u", udpPort, "-c", hosts, "&")
```

Comment:
Is the executable always at `/`? Shouldn't this be something relative to the current directory? (You may need to use `go run` instead or something…)

Reply:
There should not be a `&` at the end of the argument list; you're not spawning this through a shell, so `&` doesn't background things (it just gets passed to the application as an extra argument).

Reply:
> Is the executable always at `/`? Shouldn't this be something relative to the current directory? (You may need to use `go run` instead or something…)

The building takes place in docker file, sure, I can move things back into the `/APPDIR./